File: HomeWorks/src/Lesson6/Legs/SamsungLeg.java
package Lesson6.Legs;

public class SamsungLeg implements ILeg {
    private int price;

    public SamsungLeg() {}

    public SamsungLeg(int price){
        this.price = price;
    }

    @Override
    public void step() {
        System.out.println("Шагает нога Samsung");
    }

    public void setPrice(int price) {
        this.price = price;
    }

    @Override
    public int getPrice() {
        return price;
    }
}

File: HomeWorkKotlin/src/Robot/Legs/SamsungLeg.kt
package Robot.Legs

class SamsungLeg(private val priceParam : Int) : ILeg {

    override var price : Int = priceParam
        set(value) {
            if(value > 0) field = value
        }

    override fun step() {
        println("Шагает нога Samsung")
    }
}